### ISSUE TITLE
CS7038: 'Failed to emit module' declaring and then assigning `ReadOnlySpan<char>`

### ISSUE THREAD

**Issue by ericwj** [dotnet/roslyn#35487]
**Version Used**: 
3.100.19.20907 (01f3eb10)

**Steps to Reproduce**:

Compile this.
```C#
void FailToEmitDll(ReadOnlySpan<char> span) {
    ReadOnlySpan<char> temp;
    temp = 1 switch
    {
        _ => span
    };
}
```
**Expected Behavior**:
It compiles.

**Actual Behavior**:
`1>CSC : error CS7038: Failed to emit module 'Tests'.`

**Workaround**
Change the declaration-and-assignment of `temp` to read:
```
var temp = 1 switch (…)
```
Similar to #33275.

**Comment by jcouv** [comment]
This was fixed recently (PR https://github.com/dotnet/roslyn/pull/35311). The issue is when using a ref struct in a switch expression.
Tagging @gafter to confirm and close.

**Comment by gafter** [comment]
Verified that this is fixed by #35311
